### PR TITLE
Change how we update remote windows

### PR DESCRIPTION
**Pull request by chrisnojima**
tldr; Send things over IPC way less often, speed up startup time

The current strategy is to have the main window house redux and subscribe to all changes and send this to the main thread. 
The main thread then computes the subsets the remote windows want and sends those to individual subscribers.
Us sending the whole store over to the main thread is pretty expensive, especially when no remote cares about it

So instead

- [x] Call the selectors for the remote windows and only send when that object changes (using JSON.stringify now but could use immutable when we get that everywhere. it only takes like 2 ms to do the stringify). How this works is very naive and could be even faster and better.
- [x] Throttle the sending no matter what. This shouldn't matter much as in-between store states shouldn't matter to the remotes at all. We can adjust this value down if we want

also i noticed if you reload the browser the focus change events got broken so i'm clearing all event handlers on a reload of the code

I used console.profile around both sending strategies to do timing and also timed when the app 'settles down' (no more console output)

Using the the old way:
- [x] 58 IPC calls
- [x] Settle time ~ 27 seconds
- [x] Each IPC takes ~17ms

New way:
- [x] 9 IPC attempts (some skipped actually sending)
- [x] Settle time ~ 13 seconds (twice as fast!)
- [x] Each IPC takes ~4ms

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: desktop/renderer/index.js [hunk 1]
```diff
@@ -77,16 +79,31 @@ function setupApp (store) {
   })
 
   const currentWindow = electron.remote.getCurrentWindow()
-  currentWindow.on('focus', () => {
-    store.dispatch(changedFocus(true))
-  })
-  currentWindow.on('blur', () => {
-    store.dispatch(changedFocus(false))
-  })
+  // This fixes reload problems with stale listeners
```
Comment (chromakode) [inline desktop/renderer/index.js]: I'm a bit puzzled by this. What would cause a stale listener? Is this code running on each HMR?

Comment (chrisnojima) [inline desktop/renderer/index.js]: i'm talking about reloading the window with command+r. if you do that your old eventHandlers still exist but in a busted state.  we were suffering from this: https://github.com/electron/electron/issues/3778
